src/server/controllers/rooms.controller.js

```diff
@@ -0,0 +1,157 @@
+const { v4: uuidv4 } = require('uuid');
+
+const rooms = {};
+
+const create = (ws, data) => {
+  const room = rooms[data.room];
+
+  if (room) {
+    ws.send(JSON.stringify({ type: 'error', message: 'Room already exist' }));
+
+    return;
+  }
+
+  const newRoom = {
+    users: [{ ws, username: data.username }],
+    messages: [],
+  };
+
+  rooms[data.room] = newRoom;
+
+  ws.send(
+    JSON.stringify({
+      type: 'created',
+      room: data.room,
+      username: data.username,
+    }),
+  );
+};
+
+const join = (ws, data) => {
+  const room = rooms[data.room];
+
+  if (!room) {
+    ws.send(JSON.stringify({ type: 'error', message: 'Room does not exist' }));
+
+    return;
+  }
+
+  room.users.push({ ws, username: data.username });
+
+  ws.send(
+    JSON.stringify({
+      type: 'joined',
+      room: data.room,
+      username: data.username,
+      messages: room.messages,
+    }),
+  );
+};
+
+const rename = (ws, data) => {
+  const { room, newRoomName } = data;
+  const currentRoom = rooms[room];
+
+  if (!room || !currentRoom) {
```

Comment:
The condition `!room` should be `!currentRoom` to correctly check if the current room exists.